### ISSUE TITLE
ModuleNotFoundError: No module named 'mod_wsgi'

### ISSUE THREAD

**Issue by funcy39** [GrahamDumpleton/mod_wsgi#813]
Hello Graham,
I am using Windows system and I am doing a project with Django.
I followed the all steps in the guide to configure Apache Server and mod_wsgi. [Django's document](https://docs.djangoproject.com/zh-hans/4.1/howto/deployment/wsgi/modwsgi/) showed how to configure `httpd.conf` file in Apache.

> Basic configuration
> Once you’ve got mod_wsgi installed and activated, edit your Apache server’s [httpd.conf](https://cwiki.apache.org/confluence/display/httpd/DistrosDefaultLayout) file and add the following.
> ```
> WSGIScriptAlias / /path/to/mysite.com/mysite/wsgi.py
> WSGIPythonHome /path/to/venv
> WSGIPythonPath /path/to/mysite.com
> <Directory /path/to/mysite.com/mysite>
> <Files wsgi.py>
> Require all granted
> </Files>
> </Directory>
> ```
Starting Apache in cmd, but Apache raised a Syntax Error, "WSGIScriptAlias is not difined". I removed Apache and mod_wsgi, 
and I restarted my project with command "py manage.py runserver", the project raised an error:
```Traceback (most recent call last):
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\core\management\commands\runserver.py", line 125, in inner_run
    autoreload.raise_last_exception()
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\utils\autoreload.py", line 87, in raise_last_exception
    raise _exception[1]
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\core\management\__init__.py", line 398, in execute
    autoreload.check_errors(django.setup)()
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\apps\registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\apps\config.py", line 193, in create
    import_module(entry)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1128, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1142, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mod_wsgi'
D:\Desktop\web-projects\mysite\mysite\settings.py changed, reloading.
Watching for file changes with StatReloader
Exception in thread django-main-thread:
Traceback (most recent call last):
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\core\management\commands\runserver.py", line 125, in inner_run
    autoreload.raise_last_exception()
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\utils\autoreload.py", line 87, in raise_last_exception
    raise _exception[1]
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\core\management\__init__.py", line 398, in execute
    autoreload.check_errors(django.setup)()
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\apps\registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\django\apps\config.py", line 193, in create
    import_module(entry)
  File "C:\Users\Administrator.USER-20191017HJ\AppData\Local\Programs\Python\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1128, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1142, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mod_wsgi'
```
I can't start my project. The project didn't need mod_wsgi before. Why does the project need the mod_wsgi and can't configure the project now?

Thanks.
platform: Windows 10
python version: 3.11.0 with no vitrual environment
Apache and mod_wsgi are all uninstalled

**Comment by GrahamDumpleton** [comment]
Don't add ``mod_wsgi.server`` to ``INSTALLED_APPS``. That documentation doesn't tell you to do that and you shouldn't be adding it if manually configuring Apache.

**Comment by funcy39** [comment]
OK. The server can run now.
But how to configure Apache and mod_wsgi in Windows? I can't find a clear tutorial.

**Comment by GrahamDumpleton** [comment]
Have you looked at all at the mod_wsgi documentation?

* https://modwsgi.readthedocs.io/en/master/getting-started.html

**Comment by funcy39** [comment]
How to configure source code in Windows cmd? `./configure` does not work.
Should I see  _quick installation guide_?

- https://modwsgi.readthedocs.io/en/master/user-guides/quick-installation-guide.html#configuring-the-source-code

**Comment by GrahamDumpleton** [comment]
You have to use the `pip install mod_wsgi` method for building mod_wsgi Apache module on Windows. See:

* https://pypi.org/project/mod-wsgi/

You just can't use the `mod_wsgi server` or Django management command integration on Windows.

**Comment by funcy39** [comment]
And, the other link for Windows in documentation(https://github.com/GrahamDumpleton/mod_wsgi/blob/develop/win32/README.rst) is not exist.
What should read instead?

- https://modwsgi.readthedocs.io/en/master/installation.html

**Comment by GrahamDumpleton** [comment]
You need to use `pip install mod_wsgi` method:

* https://pypi.org/project/mod-wsgi/

as already mentioned.

There are no separate step by step Windows instructions because I don't use Windows.

So if you really need step by step instructions because you can't adapt the generic instructions, you will need to seek out some third party blog post about it.

**Comment by funcy39** [comment]
Oh. Thanks.
But I read the third party blog post after seeing documentation, they didn't work at all!
Is there anyone using Windows and installed succesfully?

**Comment by GrahamDumpleton** [comment]
Yes. If you at least have the module loading, describe what current Apache/mod_wsgi configuration you have and what the problem you are having with it is.

**Comment by funcy39** [comment]
OK.
https://github.com/GrahamDumpleton/mod_wsgi/blob/develop/README.rst recommend me use the Apache distribution from Apache Lounge (http://www.apachelounge.com/), I unpacked the file _httpd-2.4.54-win64-VS17.zip_ in D:/.
I used mod_wsgi-4.9.2-cp311-cp311-win_amd64.whl file to install mod_wsgi because I can't install with .tar.gz. I can use `mod_wsgi-express` in cmd.
Is the installation ok? If I want to run project with Apache, what should I configure?

**Comment by GrahamDumpleton** [comment]
Where did you get `mod_wsgi-4.9.2-cp311-cp311-win_amd64.whl`? I don't supply wheels. If you used file from:

* https://www.lfd.uci.edu/~gohlke/pythonlibs/

I can't help you as they are nothing to do with me.

I can only help you if you use what is described in:

* https://pypi.org/project/mod-wsgi/

which in summary for Windows is:

Run `pip install mod_wsgi`.

Then run `mod_wsgi-express module-config` and copy what is output into your Apache config file.

Then follow normal Django or mod_wsgi instructions to setup your WSGI application. I can't be more specific about this step as it depends on what WSGI framework you are using.

**Comment by funcy39** [comment]
Thanks for answering.
I am trying to install mod_wsgi in pypi. If downloading failed again, I may open another issue.
Best regards.
